pipestat/pipestat.py

```diff
@@ -460,6 +462,144 @@ def summarize(
+        stats_path = self._create_stats_summary(pipeline_name, pipeline_type)
+        objs_path = self._create_obj_summary(pipeline_name, pipeline_type)
+
+        table_path_list = [stats_path, objs_path]
+        return table_path_list
+
+    def _create_stats_summary(self, pipeline_name, pipeline_type) -> str:
+        """
+        Create stats spreadsheet and columns to be considered in the report, save
+        the spreadsheet to file
+
+        :param pipestat manager object: the project to be summarized
+        :param str pipeline_name: name of the pipeline to tabulate results for
+        :param str pipeline_type: whether the project-level pipeline results
+            should be tabulated
+        :return str: tsv_outfile_path
+        """
+        # Create stats_summary file
+        columns = set()
+        stats = []
+        _LOGGER.info("Creating stats summary")
+        if pipeline_type == "project":
+            # TODO for project level should we actually be calling project.backend.get_samples(pipeline_type='project')?
+            reported_stats = ["project_name", self.project_name or "No Project Name Supplied"]
+            results = fetch_pipeline_results(
+                project=self,
+                pipeline_name=pipeline_name,
+                sample_name=None,
+                inclusion_fun=lambda x: x not in OBJECT_TYPES,
+                pipeline_type=pipeline_type,
+            )
+            for k, v in results.items():
+                reported_stats.append(k)
+                reported_stats.append(v)
+            stats.append(reported_stats)
+            columns = ["Project", "Project Name", "Results"]
+
+        else:
+            sample_index = 0
+            for sample in self.backend.get_samples(pipeline_type="sample"):
+                sample_index += 1
+                sample_name = sample[0]
+                pipeline_type = sample[1]
+                reported_stats = [sample_index, sample_name]
+                results = fetch_pipeline_results(
+                    project=self,
+                    pipeline_name=pipeline_name,
+                    sample_name=sample_name,
+                    inclusion_fun=lambda x: x not in OBJECT_TYPES,
+                    pipeline_type=pipeline_type,
+                )
+                for k, v in results.items():
+                    reported_stats.append(k)
+                    reported_stats.append(v)
+                stats.append(reported_stats)
+                columns = ["Sample Index", "Sample Name", "Results"]
+
+        tsv_outfile_path = get_file_for_table(self, pipeline_name, "stats_summary.tsv")
+
+        stats.insert(0, columns)
+
+        with open(tsv_outfile_path, "w") as tsv_outfile:
+            writer = csv.writer(tsv_outfile, delimiter="\t")
+            for row in stats:
+                writer.writerow(row)
+
+        _LOGGER.info(
+            f"'{pipeline_name}' pipeline stats summary (n={len(stats)}):" f" {tsv_outfile_path}"
```

Comment:
why are there two strings next to one another here? why is this not a single string?